### PR TITLE
Closes #351 | Implemented dataloader for indoler

### LINKED ISSUE #351: Create dataset loader for Indoler
Dataloader name: `indoler/indoler.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?indoler

|     Dataset| indoler |
|-------------|---|
| Description | The dataset contains 993 annotated court decision documents. The document was taken from the Decision of the Supreme Court of Indonesia (https://decision3.mahkamahagung.go.id/) for 5 state courts of DKI Jakarta with a total of 1000 criminal documents cases. |
| Subsets | - |
| Languages | ind |
| Tasks | Named Entiy Recognition |
| License | Unknown (unknown) |
| Homepage | https://github.com/ir-nlp-csui/indoler/tree/main |
| HF URL | - |
| Paper URL | https://ieeexplore.ieee.org/abstract/document/9263157 |

### PR DESCRIPTION
**Pull request by luckysusanto**
Closes #351 

Note:
Despite only having 993 data instances, there are almost 5.9M tokens that need to be processed.
the testcase ran successfully, but took my laptop 150s. test case
If anyone knows how I can optimize this, please do tell me.

### Checkbox
- [X] Confirm that this PR is linked to the dataset issue.
- [X] Create the dataloader script `seacrowd/sea_datasets/indoler/indoler.py` (please use only lowercase and underscore for dataset naming).
- [X] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [X] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [X] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [X] Confirm dataloader script works with `datasets.load_dataset` function.
- [X] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/indoler/indoler.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/indoler/indoler.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+        if self.config.schema == "source":
+            features = datasets.Features({
+                "id": datasets.Value("string"),
+                "owner": datasets.Value("string"),
+                "lawyer": datasets.ClassLabel(names=[False, True]),
+                "verdict": datasets.ClassLabel(names=["guilty", "bebas", "lepas"]),
+                "indictment": datasets.ClassLabel(names=["NA", "tunggal", "subsider", "komul", "alternatif", "kombinasi", "gabungan"]),
+                "text-tags": datasets.Sequence(datasets.ClassLabel(names=
+                    ['O', 'B-Jenis Amar', 'B-Jenis Dakwaan', 'B-Jenis Perkara', 'B-Melanggar UU (Dakwaan)',
```
Comment (holylovenia) [inline seacrowd/sea_datasets/indoler/indoler.py:120]: Could we use a constant instead? Example:
```
NAMED_ENTITIES = ['O', 'B-Jenis Amar', 'B-Jenis Dakwaan', 'B-Jenis Perkara', 'B-Melanggar UU (Dakwaan)', 
                    'B-Melanggar UU (Pertimbangan Hukum)', 'B-Melanggar UU (Tuntutan)', 'B-Nama Hakim Anggota', 'B-Nama Hakim Ketua', 
                    'B-Nama Jaksa', 'B-Nama Panitera', 'B-Nama Pengacara', 'B-Nama Pengadilan', 
                    'B-Nama Saksi', 'B-Nama Terdakwa', 'B-Nomor Putusan', 'B-Putusan Hukuman', 
                    'B-Tanggal Kejadian', 'B-Tanggal Putusan', 'B-Tingkat Kasus', 'B-Tuntutan Hukuman', 
                    'I-Jenis Amar', 'I-Jenis Dakwaan', 'I-Jenis Perkara', 'I-Melanggar UU (Dakwaan)', 
                    'I-Melanggar UU (Pertimbangan Hukum)', 'I-Melanggar UU (Tuntutan)', 'I-Nama Hakim Anggota', 'I-Nama Hakim Ketua', 
                    'I-Nama Jaksa', 'I-Nama Panitera', 'I-Nama Pengacara', 'I-Nama Pengadilan', 
                    'I-Nama Saksi', 'I-Nama Terdakwa', 'I-Nomor Putusan', 'I-Putusan Hukuman', 
                    'I-Tanggal Kejadian', 'I-Tanggal Putusan', 'I-Tingkat Kasus', 'I-Tuntutan Hukuman']
...
"text-tags": datasets.Sequence(datasets.ClassLabel(names=NAMED_ENTITIES)),
```

File: seacrowd/sea_datasets/indoler/indoler.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                    'I-Jenis Amar', 'I-Jenis Dakwaan', 'I-Jenis Perkara', 'I-Melanggar UU (Dakwaan)', 
+                    'I-Melanggar UU (Pertimbangan Hukum)', 'I-Melanggar UU (Tuntutan)', 'I-Nama Hakim Anggota', 'I-Nama Hakim Ketua', 
+                    'I-Nama Jaksa', 'I-Nama Panitera', 'I-Nama Pengacara', 'I-Nama Pengadilan', 
+                    'I-Nama Saksi', 'I-Nama Terdakwa', 'I-Nomor Putusan', 'I-Putusan Hukuman', 
+                    'I-Tanggal Kejadian', 'I-Tanggal Putusan', 'I-Tingkat Kasus', 'I-Tuntutan Hukuman']
+                )),
+                "text": datasets.Sequence(datasets.Value("string")),
+            })
+        elif self.config.schema == "seacrowd_seq_label":
+            features = schemas.seq_label.features(
```
Comment (holylovenia) [inline seacrowd/sea_datasets/indoler/indoler.py:134]: Could we use a constant instead? Example:
```
NAMED_ENTITIES = ['O', 'B-Jenis Amar', 'B-Jenis Dakwaan', 'B-Jenis Perkara', 'B-Melanggar UU (Dakwaan)', 
                    'B-Melanggar UU (Pertimbangan Hukum)', 'B-Melanggar UU (Tuntutan)', 'B-Nama Hakim Anggota', 'B-Nama Hakim Ketua', 
                    'B-Nama Jaksa', 'B-Nama Panitera', 'B-Nama Pengacara', 'B-Nama Pengadilan', 
                    'B-Nama Saksi', 'B-Nama Terdakwa', 'B-Nomor Putusan', 'B-Putusan Hukuman', 
                    'B-Tanggal Kejadian', 'B-Tanggal Putusan', 'B-Tingkat Kasus', 'B-Tuntutan Hukuman', 
                    'I-Jenis Amar', 'I-Jenis Dakwaan', 'I-Jenis Perkara', 'I-Melanggar UU (Dakwaan)', 
                    'I-Melanggar UU (Pertimbangan Hukum)', 'I-Melanggar UU (Tuntutan)', 'I-Nama Hakim Anggota', 'I-Nama Hakim Ketua', 
                    'I-Nama Jaksa', 'I-Nama Panitera', 'I-Nama Pengacara', 'I-Nama Pengadilan', 
                    'I-Nama Saksi', 'I-Nama Terdakwa', 'I-Nomor Putusan', 'I-Putusan Hukuman', 
                    'I-Tanggal Kejadian', 'I-Tanggal Putusan', 'I-Tingkat Kasus', 'I-Tuntutan Hukuman']
...
features = schemas.seq_label.features(NAMED_ENTITIES),
```

File: seacrowd/sea_datasets/indoler/indoler.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+                    "idx_path": test_path,
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": data_path,
+                    "idx_path": valid_path,
+                    "split": "dev",
```
Comment (holylovenia) [inline seacrowd/sea_datasets/indoler/indoler.py:184]: `"dev"` --> `"validation"`

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @luckysusanto, thanks for your contribution! I have a few suggestions:

File: seacrowd/sea_datasets/indoler/indoler.py (hunk 3 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoler/indoler.py]: Changed!

File: seacrowd/sea_datasets/indoler/indoler.py (hunk 2 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoler/indoler.py]: Thanks for the comment! Yeah, it looks neater now ^^

File: seacrowd/sea_datasets/indoler/indoler.py (hunk 1 above)
Comment (luckysusanto) [reply seacrowd/sea_datasets/indoler/indoler.py]: Implemented!

**Comment by luckysusanto** [comment]
Requesting review!
@holylovenia

**Review by holylovenia — APPROVED** [review]
LGTM!! Thanks a lot, @luckysusanto. 😁

**Review by SamuelCahyawijaya — APPROVED** [review]
@luckysusanto : thank you for the contribution! LGTM!

**Pull request merged**